src/session/session_api.c

```diff
@@ -1703,7 +1724,17 @@ __wt_session_range_truncate(
     if (!F_ISSET(start, WT_CURSTD_KEY_INT)) {
         needs_next_prev = true;
         if (orig_start_key != NULL) {
-            WT_ERR_NOTFOUND_OK(start->search_near(start, &cmp), true);
+            ret = start->search_near(start, &cmp);
+            if (ret == WT_PREPARE_CONFLICT) {
+                /* Don't return a prepare conflict if the key we land on falls outside the truncate
+                 * range. */
+                if (__check_key_outside_truncate_range(start, trunc_info)) {
+                    F_SET(session->txn, WT_TXN_IGNORE_PREPARE);
+                    WT_ERR_NOTFOUND_OK(start->search_near(start, &cmp), true);
+                    F_CLR(session->txn, WT_TXN_IGNORE_PREPARE);
```

Comment:
I'm also curious about the scenario
```
10
20 (prepared)
30 (prepared)
40
```
and `start` == 25.

If we `search_near` and find `20` which is outside the truncation range what happens on the `WT_TXN_IGNORE_PREPARE` search_near call?

Reply:
What I think happens is that we ignore the prepare conflict error for the key 20, so we are able to position on that key without erroring out. Then on the next iteration, we'll go to the next key 30 - this should generate a prepare conflict error, but we won't ignore the error this time.

I can also add more test cases if there are specific cases you are curious about.